### PR TITLE
Avoid breaking on empty begin nodes within conditionals

### PR DESCRIPTION
**Pull request by prikha**
I stumbled upon an error: 

```
NoMethodError: undefined method `type' for nil:NilClass
lib/unparser/node_helpers.rb:31:in `n?'
lib/unparser/node_helpers.rb:66:in `block (2 levels) in <module:NodeHelpers>'
lib/unparser/generation.rb:159:in `emit_body_member'
...
```

Which is produced while unparsing a snippet of code like this: 

```
'true ? "true" : ()'
'true ? () : "false"'
```

Unfortunately on the project I'm currently working I can't guarantee/rewrite incoming code snippets, so have to handle.

### CODE REVIEW

**Comment by prikha** [comment]
Markus @mbj may I ask for your input here?

**Review by mbj — APPROVED** [review]

**Comment by prikha** [comment]
Thanks for the tip, Markus @mbj. 

I've changed it to render empty bodies. And also added a line to compare the AST without comments(as the specs were green for the previous variant). Please take a look 🙏🏻

**Comment by mbj** [comment]
Looks better, still need to run CI over it, but other than that I think this is merge ready.

@prikha What are you using unparser for?

File: spec/unit/unparser_spec.rb [hunk 1]
```diff
@@ -191,6 +191,7 @@ def assert_generates_from_ast(parser, ast_with_comments, expected)
       generated = Unparser.unparse(*ast_with_comments).chomp
       expect(generated).to eql(expected)
       ast, comments = parse_with_comments(generated)
+      expect(ast).to eq(ast_with_comments.first)
```
Comment (mbj) [inline spec/unit/unparser_spec.rb:194]: We should not use `eq` matcher as this only compares via `#==` that may coerce, `#eql?` is the better non coercing operator, so `eql` rspec syntax.

Comment (prikha) [reply spec/unit/unparser_spec.rb]: Thanks. Addressed.

**Comment by prikha** [comment]
My use case is to mutate Ruby code to insert some instrumentation.

**Comment by mbj** [comment]
> My use case is to mutate Ruby code to insert some instrumentation. And the incoming snippets are out of my control, the only thing I know it is "parseable".

Yeah, this is the exact target domain of unparser, its supposed to handle "any ruby that parses" you found a valid bug.

**Comment by prikha** [comment]
Ok, can you expand a bit on how do you trigger CI for the PR?

**Comment by mbj** [comment]
> Ok, can you expand a bit on how do you trigger CI for the PR?

probably needs edits to the `.github/workflow.yaml` file, especially the `on:` events.

Its hard for me to find any time during the next days I may be able to look into it.

**Comment by prikha** [comment]
Added a line to trigger CI on PR's, it might require your approval to be run:

image

**Comment by mbj** [comment]
Running, ty.

**Comment by prikha** [comment]
By any chance it would be possible to craft a new release version containing that fix?
@mbj

**Comment by mbj** [comment]
@prikha yes, will do so this week.

**Comment by prikha** [comment]
@mbj kind reminder

**Comment by mbj** [comment]
heh sorry, will get to it this week.

**Comment by mbj** [comment]
@prikha I've released version 0.6.12. Cheers.

**Pull request merged**